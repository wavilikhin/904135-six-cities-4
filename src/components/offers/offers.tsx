import * as React from 'react';
import { connect } from 'react-redux';
import OffersList from '../offers-list/offers-list';
import Map from '../map/map';
import { getCity, getFiltredOffers } from '../../reducer/data/selectors';
import MainEmpty from '../main-empty/main-empty';
import PlaceSorting from '../place-sorting/place-sorting';
import { OfferInfo } from '../../types';

interface Props {
  filtredOffers: OfferInfo[];
  city: string;
}

const Offers: React.FC<Props> = ({ filtredOffers, city }) => {
  return (
    <div className="cities">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">
            {filtredOffers.length} places to stay in {city}
          </b>
          <PlaceSorting />
          {filtredOffers.length === 0 ? (
            <MainEmpty city={city} />
          ) : (
            <OffersList />
          )}
        </section>
        <div className="cities__right-section">
          <section className="cities__map map">
            <Map offers={filtredOffers} />
          </section>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  city: getCity(state),
  filtredOffers: getFiltredOffers(state),
});

export { Offers };
export default connect(mapStateToProps, null)(Offers);
