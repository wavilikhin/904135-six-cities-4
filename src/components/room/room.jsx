import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Operation as DataOperation,
  ActionCreator,
} from '../../reducer/data/data.js';
import {
  getCurrentOffer,
  getCurrentOfferReviews,
  getCurrentOfferNearby,
} from '../../reducer/data/selectors.js';
import ReviewsList from '../reviews-list/reviews-list.jsx';
import ReviewForm from '../review-form/review-form.jsx';
import OfferCard from '../offer-card/offer-card.jsx';
import Map from '../map/map.jsx';
import withAddFavorites from '../../hocs/with-add-favorites/with-add-favorites.jsx';
const OfferCardWrapped = withAddFavorites(OfferCard);

class Room extends PureComponent {
  constructor(props) {
    super(props);

    this._updateReviews = this._updateReviews.bind(this);
    this._postReview = this._postReview.bind(this);
    this._updateNearby = this._updateNearby.bind(this);
    this._updateFavorites = this._updateFavorites.bind(this);
  }

  _updateReviews(id) {
    if (id === -1) {
      return;
    }
    this.props.updateReviews(id);
  }

  _postReview(hotelId, reviewData) {
    if (id === -1) {
      return;
    }
    this.props.postReview(hotelId, reviewData);
  }

  _updateNearby(id) {
    if (id === -1) {
      return;
    }
    this.props.updateNearby(id);
  }

  _updateFavorites(id) {
    if (id === -1) {
      return;
    }
    this.props.handleFavoritesUpdate(id);
  }

  componentDidMount() {
    this.props.handleCurrentOfferUpdate(this.props.offerId);
    this._updateReviews(this.props.offerId);
    this._updateNearby(this.props.offerId);
  }

  render() {
    const { defaultOffer, offer } = this.props;

    let currentOffer = {};
    Object.keys(offer).length === 0
      ? (currentOffer = defaultOffer)
      : (currentOffer = offer);

    const { reviews = [], offersNearby, favoritesIds } = this.props;

    const ratingStars = currentOffer.rating * 2 * 10;

    return (
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {currentOffer.images.map((image, i) => (
                <div className="property__image-wrapper" key={`${image}+${i}`}>
                  <img
                    className="property__image"
                    src={image}
                    alt="Photo studio"
                  />
                </div>
              ))}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {currentOffer.isPremium && (
                <div className="property__mark">
                  <span>Premium</span>
                </div>
              )}
              <div className="property__name-wrapper">
                <h1 className="property__name">{name}</h1>
                <button
                  className={`property__bookmark-button button ${
                    favoritesIds.some((fav) => fav === currentOffer.id)
                      ? 'property__bookmark-button--active'
                      : ''
                  }`}
                  type="button"
                  onClick={() => {
                    this._updateFavorites(id);
                  }}
                >
                  <svg
                    className="property__bookmark-icon"
                    width={31}
                    height={33}
                  >
                    <use xlinkHref="#icon-bookmark" />
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{ width: `${ratingStars}%` }} />
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">
                  {currentOffer.rating}
                </span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {currentOffer.type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {currentOffer.bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {currentOffer.maxAdults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">
                  â‚¬{currentOffer.priceValue}
                </b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What's inside</h2>
                <ul className="property__inside-list">
                  {currentOffer.goods.map((good, i) => (
                    <li className="property__inside-item" key={`${good}+${i}`}>
                      {good}
                    </li>
                  ))}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div
                    className={`property__avatar-wrapper user__avatar-wrapper ${
                      currentOffer.host.isPro
                        ? 'property__avatar-wrapper--pro'
                        : ''
                    }`}
                  >
                    <img
                      className="property__avatar user__avatar"
                      src={`../${currentOffer.host.avatar_url}`}
                      width={74}
                      height={74}
                      alt="Host avatar"
                    />
                  </div>
                  <span className="property__user-name">
                    {currentOffer.host.name}
                  </span>
                </div>
                <div className="property__description">
                  <p className="property__text">{currentOffer.description}</p>
                </div>
              </div>
              <section className="property__reviews reviews">
                <ReviewsList reviews={reviews} />
                <ReviewForm
                  hotelId={currentOffer.id}
                  handleSubmit={this._postReview}
                />
              </section>
            </div>
          </div>
          <section className="property__map map">
            <Map
              city={currentOffer.city}
              zoom={currentOffer.cityZoom}
              offers={offersNearby}
            />
          </section>
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">
              Other places in the neighbourhood
            </h2>
            <div className="near-places__list places__list">
              {offersNearby.map((offer, i) => {
                return (
                  <OfferCardWrapped key={`${offer.id}+${i}`} cardData={offer} />
                );
              })}
            </div>
          </section>
        </div>
      </main>
    );
  }
}

Room.defaultProps = {
  defaultOffer: {
    id: -1,
    city: '',
    city: '',
    cityZoom: 0,
    isPremium: false,
    cityCoords: [],
    image: '',
    priceValue: 0,
    name: '',
    type: '',
    coords: [],
    bedrooms: 0,
    description: '',
    goods: [],
    host: {
      isPro: false,
      avatar_url: '',
      name: '',
    },
    images: [],
    isFavorite: false,
    location: {},
    maxAdults: 0,
    rating: 0,
  },
};

const mapStateToProps = (state) => ({
  offer: getCurrentOffer(state),
  reviews: getCurrentOfferReviews(state),
  offersNearby: getCurrentOfferNearby(state),
});

const mapDispatchToProps = (dispatch) => ({
  updateReviews(id) {
    dispatch(DataOperation.updateCurrentOfferReviews(id));
  },

  postReview(hotelId, reviewData) {
    dispatch(DataOperation.postReview(hotelId, reviewData));
  },

  updateNearby(id) {
    dispatch(DataOperation.updateCurrentOfferNearby(id));
  },

  handleCurrentOfferUpdate(id) {
    dispatch(ActionCreator.updateCurrentOffer(id));
  },
});

Room.propTypes = {
  offer: PropTypes.shape({
    id: PropTypes.number,
    city: PropTypes.string,
    cityZoom: PropTypes.number,
    isPremium: PropTypes.bool,
    cityCoords: PropTypes.arrayOf(PropTypes.number),
    image: PropTypes.string,
    priceValue: PropTypes.number,
    name: PropTypes.string,
    type: PropTypes.string,
    coords: PropTypes.arrayOf(PropTypes.number),
    bedrooms: PropTypes.number,
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatar_url: PropTypes.string,
      id: PropTypes.number,
      isPro: PropTypes.bool,
      name: PropTypes.string,
    }),
    images: PropTypes.arrayOf(PropTypes.string),
    isFavorite: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
    maxAdults: PropTypes.number,
    rating: PropTypes.number,
  }),
  offerId: PropTypes.number.isRequired,
  favoritesIds: PropTypes.arrayOf(PropTypes.number),
  reviews: PropTypes.arrayOf(
    PropTypes.shape({
      comment: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired,
    }),
  ),
  updateReviews: PropTypes.func.isRequired,
  offersNearby: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      city: PropTypes.string.isRequired,
      cityZoom: PropTypes.number.isRequired,
      isPremium: PropTypes.bool.isRequired,
      cityCoords: PropTypes.arrayOf(PropTypes.number).isRequired,
      image: PropTypes.string.isRequired,
      priceValue: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
      coords: PropTypes.arrayOf(PropTypes.number).isRequired,
      bedrooms: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
      goods: PropTypes.arrayOf(PropTypes.string).isRequired,
      host: PropTypes.shape({
        avatar_url: PropTypes.string.isRequired,
        id: PropTypes.number.isRequired,
        isPro: PropTypes.bool.isRequired,
        name: PropTypes.string.isRequired,
      }),
      images: PropTypes.arrayOf(PropTypes.string).isRequired,
      isFavorite: PropTypes.bool.isRequired,
      location: PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
        zoom: PropTypes.number.isRequired,
      }),
      maxAdults: PropTypes.number.isRequired,
      rating: PropTypes.number.isRequired,
    }),
  ),
  updateNearby: PropTypes.func.isRequired,
  handleFavoritesUpdate: PropTypes.func.isRequired,
};

export { Room };
export default withAddFavorites(
  connect(mapStateToProps, mapDispatchToProps)(Room),
);
