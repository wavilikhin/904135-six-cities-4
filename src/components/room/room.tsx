import * as React from 'react';
import ReviewsList from '../reviews-list/reviews-list';
import ReviewForm from '../review-form/review-form';
import { OfferCard } from '../offer-card/offer-card';
import Map from '../map/map';
import { withAddFavorites } from '../../hocs/with-add-favorites/with-add-favorites';
import { OfferInfo, Comment, ReviewItem } from '../../types';
import { connect } from 'react-redux';
import { AppStateType } from '../../reducer/reducer';
import {
  getCurrentOfferNearby,
  getCurrentOfferReviews,
  getOffers,
} from '../../reducer/data/selectors';
import {
  Operation as DataOperation,
  ActionCreator,
} from '../../reducer/data/data';

const OfferCardWrapped = withAddFavorites(OfferCard);

type OwnPropsType = {
  offerId: number;
  favoritesIds: number[];
  handleFavoritesUpdate: (id: number) => void;
};

type StateToPropsTypes = {
  offers: OfferInfo[];
  reviews: ReviewItem[];
  offersNearby: OfferInfo[];
};

type DispatchToPropsTypes = {
  updateReviews: (id: number) => void;

  postReview: (hotelId: number, reviewData: Comment) => void;

  updateNearby: (id: number) => void;
  handleCurrentOfferUpdate: (id: number) => void;
};

type Props = OwnPropsType & StateToPropsTypes & DispatchToPropsTypes;

class Room extends React.Component<Props> {
  props: Props;

  constructor(props) {
    super(props);

    this._updateReviews = this._updateReviews.bind(this);
    this._postReview = this._postReview.bind(this);
    this._updateNearby = this._updateNearby.bind(this);
    this._updateFavorites = this._updateFavorites.bind(this);
  }

  _updateReviews(id: number): void {
    if (id === -1) {
      return;
    }
    this.props.updateReviews(id);
  }

  _postReview(hotelId: number, reviewData: Comment): void {
    if (hotelId === -1) {
      return;
    }
    this.props.postReview(hotelId, reviewData);
  }

  _updateNearby(id: number): void {
    if (id === -1) {
      return;
    }
    this.props.updateNearby(id);
  }

  _updateFavorites(id: number): void {
    if (id === -1) {
      return;
    }
    this.props.handleFavoritesUpdate(id);
  }

  componentDidMount() {
    this.props.handleCurrentOfferUpdate(Number(this.props.offerId));
    this._updateReviews(Number(this.props.offerId));
    this._updateNearby(Number(this.props.offerId));
  }

  render() {
    // TODO: Если оффер не найден по номеру показывать заглушку
    let { offers, reviews, offersNearby, favoritesIds, offerId } = this.props;

    console.log(offersNearby);

    let currentOffer = offers.find(
      (offer) => Number(offer.id) === Number(offerId),
    );

    let ratingStars: number;
    currentOffer
      ? (ratingStars = currentOffer.raiting * 2 * 10)
      : (ratingStars = 0);

    return (
      <main className="page__main page__main--property">
        {currentOffer ? (
          <>
            <section className="property">
              <div className="property__gallery-container container">
                <div className="property__gallery">
                  {currentOffer.images.map((image, i) => (
                    <div
                      className="property__image-wrapper"
                      key={`${image}+${i}`}
                    >
                      <img
                        className="property__image"
                        src={image}
                        alt="Photo studio"
                      />
                    </div>
                  ))}
                </div>
              </div>
              <div className="property__container container">
                <div className="property__wrapper">
                  {currentOffer.isPremium && (
                    <div className="property__mark">
                      <span>Premium</span>
                    </div>
                  )}
                  <div className="property__name-wrapper">
                    <h1 className="property__name">{currentOffer.name}</h1>
                    <button
                      className={`property__bookmark-button button ${
                        favoritesIds.some((fav) => fav === currentOffer.id)
                          ? 'property__bookmark-button--active'
                          : ''
                      }`}
                      type="button"
                      onClick={() => {
                        this._updateFavorites(currentOffer.id);
                      }}
                    >
                      <svg
                        className="property__bookmark-icon"
                        width={31}
                        height={33}
                      >
                        <use xlinkHref="#icon-bookmark" />
                      </svg>
                      <span className="visually-hidden">To bookmarks</span>
                    </button>
                  </div>
                  <div className="property__rating rating">
                    <div className="property__stars rating__stars">
                      <span style={{ width: `${ratingStars}%` }} />
                      <span className="visually-hidden">Rating</span>
                    </div>
                    <span className="property__rating-value rating__value">
                      {currentOffer.raiting}
                    </span>
                  </div>
                  <ul className="property__features">
                    <li className="property__feature property__feature--entire">
                      {currentOffer.type}
                    </li>
                    <li className="property__feature property__feature--bedrooms">
                      {currentOffer.bedrooms} Bedrooms
                    </li>
                    <li className="property__feature property__feature--adults">
                      Max {currentOffer.maxAdults} adults
                    </li>
                  </ul>
                  <div className="property__price">
                    <b className="property__price-value">
                      €{currentOffer.priceValue}
                    </b>
                    <span className="property__price-text">&nbsp;night</span>
                  </div>
                  <div className="property__inside">
                    <h2 className="property__inside-title">What's inside</h2>
                    <ul className="property__inside-list">
                      {currentOffer.goods.map((good, i) => (
                        <li
                          className="property__inside-item"
                          key={`${good}+${i}`}
                        >
                          {good}
                        </li>
                      ))}
                    </ul>
                  </div>
                  <div className="property__host">
                    <h2 className="property__host-title">Meet the host</h2>
                    <div className="property__host-user user">
                      <div
                        className={`property__avatar-wrapper user__avatar-wrapper ${
                          currentOffer.host.isPro
                            ? 'property__avatar-wrapper--pro'
                            : ''
                        }`}
                      >
                        <img
                          className="property__avatar user__avatar"
                          src={`../${currentOffer.host.avatar_url}`}
                          width={74}
                          height={74}
                          alt="Host avatar"
                        />
                      </div>
                      <span className="property__user-name">
                        {currentOffer.host.name}
                      </span>
                    </div>
                    <div className="property__description">
                      <p className="property__text">
                        {currentOffer.description}
                      </p>
                    </div>
                  </div>
                  <section className="property__reviews reviews">
                    <ReviewsList reviews={reviews} />
                    <ReviewForm
                      hotelId={currentOffer.id}
                      handleSubmit={this._postReview}
                    />
                  </section>
                </div>
              </div>
              <section className="property__map map">
                <Map offers={offersNearby} />
              </section>
            </section>
            <div className="container">
              <section className="near-places places">
                <h2 className="near-places__title">
                  Other places in the neighbourhood
                </h2>
                <div className="near-places__list places__list">
                  {offersNearby.map(
                    (offer: OfferInfo, i: number): React.ReactElement => {
                      return (
                        <OfferCardWrapped
                          key={`${offer.id}+${i}`}
                          cardData={offer}
                        />
                      );
                    },
                  )}
                </div>
              </section>
            </div>
          </>
        ) : (
          <>'No room'</>
        )}
      </main>
    );
  }
}

const mapStateToProps = (state: AppStateType) => ({
  offers: getOffers(state),
  reviews: getCurrentOfferReviews(state),
  offersNearby: getCurrentOfferNearby(state),
});

const mapDispatchToProps = (dispatch) => ({
  updateReviews(id: number) {
    dispatch(DataOperation.getOfferComments(id));
  },

  postReview(hotelId: number, reviewData: Comment) {
    dispatch(DataOperation.postReview(hotelId, reviewData));
  },

  updateNearby(id) {
    dispatch(DataOperation.updateOfferNearby(id));
  },

  handleCurrentOfferUpdate(id) {
    dispatch(ActionCreator.updateCurrentOffer(id));
  },
});

export { Room };

export default connect(mapStateToProps, mapDispatchToProps)(Room);
