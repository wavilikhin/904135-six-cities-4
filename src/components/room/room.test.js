import React from 'react';
import renderer from 'react-test-renderer';
import { OFFERS } from '../../test/__mocks__/offers';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space.js';
import { Room } from './room.jsx';

const mockStore = configureStore([]);
const offerInfo = OFFERS[0];

describe(`Room component snapshot test`, () => {
  it(`Should render room offer page`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        offers: [],
      },
      [NameSpace.STATE]: {
        city: '',
      },
      [NameSpace.USER]: {
        authStatus: 'NO_AUTH',
        userEmail: '',
        userFavorites: [],
      },
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <Room offerInfo={offerInfo} offerId={1} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
