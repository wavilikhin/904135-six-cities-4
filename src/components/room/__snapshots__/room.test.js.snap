// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room component snapshot test Should render room offer page 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            
          </h1>
          <button
            className="property__bookmark-button button property__bookmark-button--active"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height={33}
              width={31}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "48%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            2.4
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            house
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            5
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            6
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            813
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height={74}
                src="../img/avatar-angelina.jpg"
                width={74}
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              0
            </span>
          </h2>
          <ul
            className="reviews__list"
          />
          <form
            action="#"
            className="reviews__form form"
            method="post"
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                className="form__rating-input visually-hidden"
                defaultValue={5}
                id="5-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                defaultValue={4}
                id="4-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="good"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                defaultValue={3}
                id="3-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="not bad"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                defaultValue={2}
                id="2-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="badly"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                defaultValue={1}
                id="1-star"
                name="rating"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-star"
                title="terribly"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              defaultValue=""
              id="review"
              name="review"
              onChange={[Function]}
              placeholder="Tell how was your stay, what you like and what can be improved"
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set
                 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                onClick={[Function]}
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <div
        id="map"
        style={
          Object {
            "height": "100%",
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <article
          className="cities__place-card place-card"
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  238
                </b>
                <span
                  className="place-card__price-text"
                >
                   
                  ∕ night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button "
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "62%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/2"
                id="linkToOffer"
                onClick={[Function]}
              >
                Amazing and Extremely Central Flat
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
        <article
          className="cities__place-card place-card"
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  722
                </b>
                <span
                  className="place-card__price-text"
                >
                   
                  ∕ night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button "
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "88%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/3"
                id="linkToOffer"
                onClick={[Function]}
              >
                The Pondhouse - A Magical Place
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              house
            </p>
          </div>
        </article>
        <article
          className="cities__place-card place-card"
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  309
                </b>
                <span
                  className="place-card__price-text"
                >
                   
                  ∕ night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button "
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "92%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/4"
                id="linkToOffer"
                onClick={[Function]}
              >
                The Joshua Tree House
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
